File: csharp/log-levels/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        var idx = logLine.IndexOf(':');
        return logLine[(idx + 1)..]
            .Trim();
    }

    public static string LogLevel(string logLine)
    {
        var idx = logLine.IndexOf(':');
        return logLine[..idx]
            .Trim()
            .Replace('[', ' ')
            .Replace(']', ' ')
            .Trim()
            .ToLower();
    }

    public static string Reformat(string logLine) => $"{Message(logLine)} ({LogLevel(logLine)})";
}

File: fsharp/log-levels/LogLevels.fs
module LogLevels

let message (logLine: string): string =
    let idx = logLine.IndexOf(":")
    logLine[idx+2..].Trim()

let logLevel(logLine: string): string =
    let idx2 = logLine.IndexOf("]")
    logLine[1..idx2-1].ToLower()

let reformat(logLine: string): string =
    message logLine + sprintf " (%s)" (logLevel logLine)
